pandas/tests/indexing/test_loc.py

```diff
@@ -2869,6 +2869,13 @@ def test_loc_setitem_using_datetimelike_str_as_index(fill_val, exp_dtype):
     tm.assert_index_equal(df.index, expected_index, exact=True)
 
 
+def test_loc_set_int_dtype():
+    df = pd.DataFrame([list("abc")])
+    df.loc[:, "col1"] = int(5)
+
+    assert str(df.dtypes["col1"]) == "int64"
```

Comment:
Generally we want to compare the complete `DataFrame` result which will check the dtype e.g.

```
df.loc[:, "col1"] = 5
expected = DataFrame(...)
tm.assert_frame_equal(df, expected)
```